lib/dns.js

```diff
@@ -285,41 +285,38 @@ exports.getServers = function() {
 exports.setServers = function(servers) {
   // cache the original servers because in the event of an error setting the
   // servers cares won't have any servers available for resolution
-  var orig = cares.getServers();
+  const orig = cares.getServers();
 
-  var newSet = [];
+  const newSet = servers.map(serv => {
+    var ipVersion = isIp(serv);
 
-  servers.forEach(function(serv) {
-    var ver = isIp(serv);
+    if (ipVersion !== 0)
+      return [ver, serv];
 
-    if (ver)
-      return newSet.push([ver, serv]);
-
-    var match = serv.match(/\[(.*)\](:\d+)?/);
+    const testIpv6InBrackets = serv.match(/\[(.*)\](:\d+)?/);
 
-    // we have an IPv6 in brackets
-    if (match) {
-      ver = isIp(match[1]);
-      if (ver)
-        return newSet.push([ver, match[1]]);
+    if (testIpv6InBrackets) {
```

Comment:
Doesn't look like a good name to me. Perhaps `isIpv6InBrackets` would be a bit better.
No strong opinion here, though.